src/bindings/python/src/openvino/frontend/tensorflow/graph_iterator.py

```diff
@@ -0,0 +1,70 @@
+# Copyright (C) 2018-2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+# flake8: noqa
+# mypy: ignore-errors
+
+import tensorflow as tf
+from openvino.frontend.tensorflow.py_tensorflow_frontend import _FrontEndPyGraphIterator as GraphIterator
+
+from .node_decoder import TFGraphNodeDecoder
+
+
+class GraphIteratorTFGraph(GraphIterator):
+    def __init__(self, tf_graph, inner_graph=False):
+        GraphIterator.__init__(self)
+        self.m_graph = tf_graph
+        self.m_node_index = 0
+        self.m_decoders = []
+        self.m_inner_graph = inner_graph
+
+        for op in tf_graph.get_operations():
+            self.m_decoders.append(TFGraphNodeDecoder(op, inner_graph))
+
+        self.m_iterators = {}
+        for func_name, _ in self.m_graph._functions.items():
+            self.m_iterators[func_name] = None
+
+    def get_input_names(self) -> list:
+        inp_ops = filter(lambda op: op.type == "Placeholder" and len(op.inputs) == 0, self.m_graph.get_operations())
+        inp_names = []
+        for inp in inp_ops:
+            type_attr = inp.node_def.attr["dtype"].type
+            if tf.dtypes.DType(type_attr).name != "resource" or self.m_inner_graph:
+                inp_names.append(inp.name)
+        return inp_names
+
+    def get_output_names(self) -> list:
+        non_outputs = []
+        for op in self.m_graph.get_operations():
+            for inp in op.inputs:
+                non_outputs.append(inp.op.name)
+
+        outputs = []
+        for op in self.m_graph.get_operations():
+            if op.name not in non_outputs:
+                for output in op.outputs:
+                    outputs = [output.name] + outputs
+        return outputs
+
+    def is_end(self) -> bool:
+        return self.m_node_index >= len(self.m_decoders)
+
+    def reset(self):
+        self.m_node_index = 0
+
+    def size(self) -> int:
+        return len(self.m_decoders)
+
+    def next_impl(self):
```

Comment:
`next_impl`?

Reply:
`next` name cannot be used for Python method, as it shadows Python `next` operator. So I made a separate method `next_impl()` for Python binding, while in c++ we have `next()`, which calls `next_impl()`.